File: server/php/src/Application/Interfaces/Services/PaymentLinkServiceInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\Services;

use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\ResponseDto;

interface PaymentLinkServiceInterface
{
    public function createPaymentLink(RequestDto $requestDto): ResponseDto;
}

File: server/python/app/application/interfaces/services/payment_link_service_interface.py
from abc import ABC, abstractmethod
from app.application.dtos.payment_link.request_dto import CreatePaymentLinkRequestDto
from app.application.dtos.payment_link.response_dto import CreatePaymentLinkResponseDto

class PaymentLinkServiceInterface(ABC):

    @abstractmethod
    def create_payment_link(self, request_dto: CreatePaymentLinkRequestDto) -> CreatePaymentLinkResponseDto:
        pass